src/main/java/seedu/address/model/item/DateValidator.java

```diff
@@ -0,0 +1,22 @@
+package seedu.address.model.item;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class DateValidator {
+    private final DateTimeFormatter dateFormatter;
+
+    public DateValidator(DateTimeFormatter dateFormatter) {
+        this.dateFormatter = dateFormatter;
+    }
+
+    public boolean isValid(String dateStr) {
+        try {
```

Comment:
Would it be better if we change this to isParsableToDate or isParsable?
I am thinking that isValid sounds like it already has done the validation for everything. 
E.g. If the year is 1000000 it could be parsable but it is not valid.